h2o-core/src/main/java/hex/Model.java

```diff
@@ -1157,24 +1174,36 @@ public double lift_top_group() {
     final String response = parms._response_column;
 
     // whether we need to be careful with categorical encoding - the test frame could be either in original state or in encoded state
-    final boolean checkCategoricals =
-        parms._categorical_encoding == Parameters.CategoricalEncodingScheme.OneHotExplicit ||
-        parms._categorical_encoding == Parameters.CategoricalEncodingScheme.Eigen ||
-        parms._categorical_encoding == Parameters.CategoricalEncodingScheme.Binary;
+    // keep in sync with FrameUtils.categoricalEncoder: as soon as a categorical column has been encoded, we should check here.
+    final boolean checkCategoricals = Arrays.asList(
+            Parameters.CategoricalEncodingScheme.Binary,
+            Parameters.CategoricalEncodingScheme.LabelEncoder,
+            Parameters.CategoricalEncodingScheme.Eigen,
+            Parameters.CategoricalEncodingScheme.EnumLimited,
+            Parameters.CategoricalEncodingScheme.OneHotExplicit
+    ).indexOf(parms._categorical_encoding) >= 0;
+
+    final boolean multiColumnsEncoding = Arrays.asList(
+            Parameters.CategoricalEncodingScheme.Binary,
+            Parameters.CategoricalEncodingScheme.OneHotExplicit
+    ).indexOf(parms._categorical_encoding) >= 0;
 
     // test frame matches the user-given frame (before categorical encoding, if applicable)
     if( checkCategoricals && origNames != null ) {
-      boolean match = Arrays.equals(origNames, test._names);
+      boolean match = Arrays.equals(origNames, test.names());
       if (!match) {
-        match = true;
         // In case the test set has extra columns not in the training set - check that all original pre-encoding columns are available in the test set
         // We could be lenient here and fill missing columns with NA, but then it gets difficult to decide whether this frame is pre/post encoding, if a certain fraction of columns mismatch...
```

Comment:
@michalkurka : do you understand this comment?
I don't understand why we can't be lenient indeed...  we should be able to make predictions on a test frame having only a subset of predictors columns (origNames), I don't see why we require them to be all present in the test frame.
any idea?